docs/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.md

```diff
@@ -0,0 +1,107 @@
+  secrets:
+    api_key: <user-input> #  Secret type. The api key get from "https://xxx.com". Don't replace the '<user-input>' placeholder. The application will prompt you to enter a value when it runs.
+  ```
+
+## Develop a flow using package tools with custom strong type connections
+To develop a flow using package tools with custom strong type connections, follow these steps:
+* Step1: Refer to the [Create and Use Tool Package](create-and-use-tool-package.md#create-custom-tool-package) to build and install your tool package in your local environment.
+  > [!Note] Once the new tool package is installed in your local environment, a window reload is necessary. This action ensures that the new tools and custom strong type connections become visible and accessible.
+
+* Step2: Develop a flow with custom tools. Please take [this folder](https://github.com/microsoft/promptflow/tree/main/examples/tools/use-cases/custom-strong-type-connection-package-tool-showcase) as an example.
+
+* Step3: Create a custom strong type connection using one of the following methods:
+  - If the connection type hasn't been created previously, click the 'Add connection' button to create the connection.
+    ![create_custom_strong_type_connection_in_node_interface](../../media/how-to-guides/develop-a-tool/create_custom_strong_type_connection_in_node_interface.png)
+  - Click the 'Create connection' plus sign in the CONNECTIONS section.
+    ![create_custom_strong_type_connection_add_sign](../../media/how-to-guides/develop-a-tool/create_custom_strong_type_connection_add_sign.png)
+  - Click 'Create connection' plus sign in the Custom category.
+    ![create_custom_strong_type_connection_in_custom_category](../../media/how-to-guides/develop-a-tool/create_custom_strong_type_connection_in_custom_category.png)
+
+  Fill in the `values` starting with `to-replace-with` in the connection template.
+  ![custom_strong_type_connection_template](../../media/how-to-guides/develop-a-tool/custom_strong_type_connection_template.png)
+
+* Step4: Use the created custom strong type connection in your flow and run.
+  ![use_custom_strong_type_connection_in_flow](../../media/how-to-guides/develop-a-tool/use_custom_strong_type_connection_in_flow.png)
+
+## Develop a flow using script tools with custom strong type connections
+To develop a flow using script tools with custom strong type connections, follow these steps:
+* Step1: Develop a flow with python script tools. Please take [this folder](https://github.com/microsoft/promptflow/tree/main/examples/tools/use-cases/custom-strong-type-connection-script-tool-showcase) as an example.
+* Step2: Using a custom strong type connection in a script tool is slightly different from using it in a package tool. When creating the connection, you will create a `CustomConnection`. Fill in the `keys` and `values` in the connection template.
+  ![custom](../../media/how-to-guides/develop-a-tool/custom_connection_template.png)
```

Comment:
It might be better to update the pic. For the connection key, use "-" or "_". I am not sure if the space is supported for the value.